xml_converter/src/string_helper.cpp

```diff
@@ -314,3 +315,16 @@ std::string long_to_hex_string(uint64_t number) {
 
     return hex_string;
 }
+
+void combine_sets(
+    std::set<std::string>* set_a,
+    std::set<std::string>* set_b,
+    std::vector<std::string>* duplicates) {
+    for (string str : *set_a) {
+        if (auto search = set_b->find(str); search != set_b->end()) {
+            duplicates->push_back(str);
+        }
+        else
+            set_b->insert(str);
+    }
+}
```

Comment:
I don't think this function belongs in string_helper. It looks specific to the category name de-duplication.


If the name of the function is descriptive of what it is supposed to do, I think you can do:
```c++
set_a.insert(set_b.begin(), set_b.end())
```

If you are trying to find the intersection of the two sets instead you could use: https://en.cppreference.com/w/cpp/algorithm/set_intersection it seems.

Otherwise maybe this function should get a different name and live closer to where it is used.